### PR TITLE
Handle invalid chunk size in ManagedHandler

### LINKED ISSUE #23143: ManagedHandler: Size overflow needs to be handled in ParseHexSize
We should just see how WinHttpHandler and CurlHandler handle overflow here, and do the same.

### PR DESCRIPTION
**Pull request by stephentoub**
To match WinHttpHandler, support chunk sizes up to ulong.MaxValue in size, and throw exceptions for anything larger.

Fixes #23143 
cc: @geoffkizer

### CODE REVIEW

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs [hunk 1]
```diff
@@ -1091,6 +1091,37 @@ public void GetAsync_ServerNeedsAuthAndNoCredential_StatusCodeUnauthorized()
...
+        [InlineData("")] // missing size
+        [InlineData("10000000000000000")] // overflowing size
+        public async Task GetAsync_InvalidChunkSize_ThrowsHttpRequestException(string chunkSize)
+        {
+            await LoopbackServer.CreateServerAsync(async (server, url) =>
+            {
+                using (var client = new HttpClient())
+                {
+                    string partialResponse = "HTTP/1.1 200 OK\r\n" +
+                        "Transfer-Encoding: chunked\r\n" +
```
Comment (davidsh) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Best practice is to always have a Date: response header in here.  I believe this is a MUST per RFC.  The other HTTP tests that use loopback server do that.

**Comment by stephentoub** [comment]
@dotnet/dnceng, we seem to be having more frequent macOS issues again, e.g.
https://ci3.dot.net/job/dotnet_corefx/job/master/job/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/2221/consoleText
```
Running on dci-mac-build-105 in /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest
[Pipeline] {
[Pipeline] step
[WS-CLEANUP] Deleting project workspace...[WS-CLEANUP] done
[Pipeline] isUnix
[Pipeline] sh
[Pipeline] }
[Pipeline] // node
[Pipeline] End of Pipeline
Setting status of 82a127bd4700f6c73c051d05c8ea3e95eed1a5db to FAILURE with url https://ci3.dot.net/job/dotnet_corefx/job/master/job/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/2221/ and message: 'Build finished. '
Using context: OSX x64 Debug Build
java.nio.channels.ClosedChannelException
	at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer.onReadClosed(ChannelApplicationLayer.java:208)
	at org.jenkinsci.remoting.protocol.ApplicationLayer.onRecvClosed(ApplicationLayer.java:222)
	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.onRecvClosed(ProtocolStack.java:832)
	at org.jenkinsci.remoting.protocol.FilterLayer.onRecvClosed(FilterLayer.java:287)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.onRecvClosed(SSLEngineFilterLayer.java:181)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.switchToNoSecure(SSLEngineFilterLayer.java:283)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processWrite(SSLEngineFilterLayer.java:503)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.processQueuedWrites(SSLEngineFilterLayer.java:248)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doSend(SSLEngineFilterLayer.java:200)
	at org.jenkinsci.remoting.protocol.impl.SSLEngineFilterLayer.doCloseSend(SSLEngineFilterLayer.java:213)
	at org.jenkinsci.remoting.protocol.ProtocolStack$Ptr.doCloseSend(ProtocolStack.java:800)
	at org.jenkinsci.remoting.protocol.ApplicationLayer.doCloseWrite(ApplicationLayer.java:173)
	at org.jenkinsci.remoting.protocol.impl.ChannelApplicationLayer$ByteBufferCommandTransport.closeWrite(ChannelApplicationLayer.java:311)
	at hudson.remoting.Channel.close(Channel.java:1295)
	at hudson.slaves.ChannelPinger$1.onDead(ChannelPinger.java:180)
	at hudson.remoting.PingThread.ping(PingThread.java:130)
	at hudson.remoting.PingThread.run(PingThread.java:86)
```
@dotnet-bot test OSX x64 Debug Build please

File: src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs (hunk 1 above)
Comment (stephentoub) [inline src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs]: Ok.  I just copied this from the above test, which doesn't have one either:
https://github.com/dotnet/corefx/pull/23160/files/82a127bd4700f6c73c051d05c8ea3e95eed1a5db#diff-06a6483ff6bb478e06f199d69522bcd4L1070
I'll leave it like this for now and we can subsequently scrub through them to add the header where it's needed.

**Comment by geoffkizer** [comment]
LGTM

**Pull request merged**